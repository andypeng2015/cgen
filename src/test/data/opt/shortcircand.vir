module foo

export function w $foo(l %a, l %i) {
@start:
  %k = slt.l %i, 3_l
  br %k, @cont, @join
@cont:
  %k = sge.l %i, 0_l
  jmp @join
@join:
  br %k, @ok, @bad
@ok:
  %s = mul.l %i, 4_l
  %a = add.l %a, %s
  %v = ld.w %a
  ret %v
@bad:
  ret 0xffffffff_w
}
