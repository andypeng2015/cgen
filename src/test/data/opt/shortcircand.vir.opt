module foo

export function w $foo(l %a, l %i) {
@2:
  %5 = slt.l %i, 0x3_l ; @17
  br %5, @3, @12(0xffffffff_w)
@3:
  %6 = sge.l %i, 0x0_l ; @18
  br %6, @7, @12(0xffffffff_w)
@7:
  %7 = lsl.l %i, 0x2_l ; @19
  %8 = add.l %a, %7 ; @20
  %v.1 = ld.w %8 ; @11
  jmp @12(%v.1)
@12(%0):
  ret %0
}
