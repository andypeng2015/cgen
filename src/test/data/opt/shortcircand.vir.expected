module foo

export function w $foo(l %a, l %i) {
@0:
  %1 = slt.l %i, 0x3_l ; @11
  br %1, @1, @10(0xffffffff_w)
@1:
  %2 = sge.l %i, 0x0_l ; @12
  br %2, @5, @10(0xffffffff_w)
@5:
  %3 = lsl.l %i, 0x2_l ; @13
  %4 = add.l %a, %3 ; @14
  %v.1 = ld.w %4 ; @9
  jmp @10(%v.1)
@10(%0):
  ret %0
}
